assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -297,16 +294,21 @@ pub fn u32rotr(
 ) -> Result<Option<CodeBlock>, AssemblyError> {
     match (imm, op_mode) {
         (Some(imm), U32OpMode::Checked) => {
-            // Assert the top of the stack is a u32 value.
-            // NOTE: We cannot use U32Assert2 since we are potentially pushing a number larger
-            // than u32 for b.
-            // TODO: double check this logic: we shouldn't use U32MUL if both inputs are not
-            // guaranteed to be u32 values
-            validate_param(imm, 0, MAX_U32_ROTATE_VALUE)?;
-            span.push_ops([Pad, U32assert2, Drop, Push(Felt::new(1 << (32 - imm)))]);
+            // if rotation is performed by 0, just verify that stack top is u32
+            if imm == 0 {
+                span.push_ops([Pad, U32assert2, Drop]);
+                return Ok(None);
+            }
+            validate_param(imm, 1, MAX_U32_ROTATE_VALUE)?;
+            span.push_ops([Push(Felt::new(1 << (32 - imm))), U32assert2]);
         }
         (Some(imm), U32OpMode::Unchecked) => {
-            validate_param(imm, 0, MAX_U32_ROTATE_VALUE)?;
+            // if rotation is performed by 0, do nothing (Noop)
```

Comment:
I think its more readable to create a dedicated variant on the match pattern instead of opening a new level

```rust
(Some(imm), U32OpMode::Unchecked) if imm == 0 => span.push_op(Noop),
(Some(imm), U32OpMode::Unchecked) => (..),
```